src/ModelContextProtocol.Analyzers/XmlToDescriptionGenerator.cs

```diff
@@ -0,0 +1,386 @@
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Text;
+using System.Collections.Immutable;
+using System.Text;
+using System.Xml.Linq;
+
+namespace ModelContextProtocol.Analyzers;
+
+/// <summary>
+/// Source generator that creates [Description] attributes from XML comments
+/// for partial methods tagged with MCP attributes.
+/// </summary>
+[Generator]
+public sealed class XmlToDescriptionGenerator : IIncrementalGenerator
+{
+    private const string GeneratedFileName = "ModelContextProtocol.Descriptions.g.cs";
+
+    public void Initialize(IncrementalGeneratorInitializationContext context)
+    {
+        // Filter method declarations with attributes. We're looking for attributed partial methods.
+        var methodModels = context.SyntaxProvider
+            .CreateSyntaxProvider(
+                static (s, _) => s is MethodDeclarationSyntax { AttributeLists.Count: > 0 } method && method.Modifiers.Any(SyntaxKind.PartialKeyword),
```

Comment:
I'm seeing a couple of issues with this approach:

1. The way this is structured today means that the generator is going to silently skip methods without a partial keyword. 
2. We're performing expensive semantic model computations for every method declaration in the compilation that is declared partial. We should try to filter down declarations on the syntactic level first (e.g. by searching for particular attributes either on the type level or method level) via the `ForAttributeWithMetadataName` helper.